packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -755,4 +754,47 @@ end
           'Unable to create XCFramework: ${xcframeworkResult.stderr}');
     }
   }
+
+  Future<void> _buildAotFrameworkWithAssemble(
```

Comment:
This doesn't need to be a separate method from `_produceAotAppFrameworkIfNeeded`, just combine them.